easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -1803,6 +1803,25 @@ def asdict(self):
             res[key] = value
         return res
 
+    def get_cuda_cc_template_value(self, key):
+        """
+        Get template value based on --cuda-compute-capabilities EasyBuild configuration option
+        and cuda_compute_capabilities easyconfig parameter.
+        Returns user-friendly error message in case neither are defined,
+        or if an unknown key is used.
+        """
+        if key.startswith('cuda_') and key in [x for (x, _) in TEMPLATE_NAMES_DYNAMIC]:
+            if key in self.template_values:
+                return self.template_values[key]
+            else:
+                error_msg = "(get_cuda_cc_template_value) Template value '%s' is not defined!\n"
```

Comment:
As this is a user facing error: I would not include this line in the error, maybe just log it and shorten it a bit: "You need to define --cuda-compute-capabilities and/or set the cuda_compute_capabilities EasyConfig parameter to a default value"

Reply:
I would definitely include the "`Template value 'cuda_xxx' is not defined`" part, since that explain what goes wrong. The 2nd part of the error message explains how to fix the issue. Both bits are relevant information for the end user.

I agree that including `(get_cuda_cc_template_value)` might be overkill though, I'll kick that out.